pkg/k8s/api.go

```diff
@@ -82,6 +92,20 @@ func NewAPI(configPath, kubeContext string, impersonate string, impersonateGroup
 		TsClient:      tsClient,
 		DynamicClient: dynamicClient,
 	}, nil
+
+}
+
+// NewAPI validates a Kubernetes config and returns a client for accessing the
+// configured cluster.
+func NewAPI(configPath, kubeContext string, impersonate string, impersonateGroup []string, timeout time.Duration) (*KubernetesAPI, error) {
```

Comment:
Is this function used anywhere? I see only calls to `newK8sAPI` and subsequent calls to `NewAPIForConfig` in the modified files where `NewAPI` was previously called.

Reply:
Keep NewAPI for compatibility, add NewAPIConfig for more extensibility